roxctl/common/config/config.go

```diff
@@ -0,0 +1,139 @@
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/env"
+	"gopkg.in/yaml.v3"
+)
+
+// Store provides the ability to read / write configurations for roxctl from / to a configuration file.
+//
+//go:generate mockgen-wrapper
+type Store interface {
+	Write(cfg *RoxctlConfig) error
+	Read() (*RoxctlConfig, error)
+}
+
+// RoxctlConfig contains all configurations available for roxctl.
+type RoxctlConfig struct {
+	CentralConfigs CentralConfigs `json:"centrals"`
+}
+
+// GetCentralConfigs retrieves all central configs. In case RoxctlConfig is nil, nil will be returned.
+func (r *RoxctlConfig) GetCentralConfigs() CentralConfigs {
+	if r == nil {
+		return nil
+	}
+	return r.CentralConfigs
+}
+
+// CentralConfigs is the list of configurations per central.
+type CentralConfigs map[string]*CentralConfig
```

Comment:
I'm not a fan of `string` as a key. Later in the code its called host but we pass there `url.String()` how about having a dedicated type for this? Another thing is the same instance could be accessed with different connection string (e.g. hostname, IP). Ideally we should use some id thats is stable between restarts (not sure if we have something like this) or certificate fingerprint as it will handle both hostnames and IP connection strings.
On the other hand maybe it's not a problem as in the worst case scenario user will need to log again.

Reply:
Hmm, I can create a separate type for this, but I don't think it's worthwhile coming up with a "ID that's stable between restarts and ways of accessing central", the same point could be made when the certificate is changed for the loadbalancer that everyone has to re-login as well, and I guess calling `roxctl central login` once more doesn't seem to be too much pain IMO.

At the end of the day, I'd expect people to use a stable endpoint to access roxctl, that'd be the public endpoint (i.e. in the case of ACSCS, it's never going to change, and most likely also not for on-prem installations). What we currently do sometimes (port-forwarding central to localhost:<some-port>) is rather unusual for customers to be doing, unless they are admins of some sort.

Let me add a type alias for this, and let's keep the hostname for now?